apps/web/src/components/auth/reset-form.tsx

```diff
@@ -0,0 +1,100 @@
+"use client";
+
+import { useEffect, useState, useTransition } from "react";
+import { useSearchParams } from "next/navigation";
+import { resetPassword, searchForToken } from "@/actions/auth";
+import { resetSchema } from "@/lib/schema/auth";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Loader2Icon } from "lucide-react";
+import { useForm } from "react-hook-form";
+import type { z } from "zod";
+
+import { Button, Form, FormControl, FormDescription, FormField, FormItem, FormMessage, Input } from "@energyleaf/ui";
+import { useToast } from "@energyleaf/ui/hooks";
+
+export default function ResetForm() {
+    const [loading, setLoading] = useState<boolean>(false);
+    const [hasToken, setToken] = useState<boolean>(false);
+    const [isPending, startTransition] = useTransition();
+    const [error, setError] = useState<string>("");
+    const { toast } = useToast();
+    const form = useForm<z.infer<typeof resetSchema>>({
+        resolver: zodResolver(resetSchema),
+        defaultValues: {
+            password: "",
+            passwordRepeat: "",
+        },
+    });
+
+    const searchParams = useSearchParams();
+
+    useEffect(() => {
+        (async () => {
+            const x = await searchForToken(searchParams.get("token"));
```

Comment:
Den Token und besser in einer Server Component fetchen und an die Client Component übergeben. Wir wollen Datenbank abfragen aus Client Components vermeiden. Auch wenn die Funktion auf dem Server läuft